### PR TITLE
lightning: replace grpc gzip compressor with klauspost/compress/gzip

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/41970

Problem Summary:

The default gzip library has poor performance.  See the issue for more details.

### What is changed and how it works?

Replace gzip compressor implementation with [klauspost/compress/gzip](https://github.com/klauspost/compress). Test the same 562GiB data in the issue, with compression enabled, the time cost reduced to 50m7s, which is a big improvement.

Also in unit test, we can see that our custom `GzipCompressor` is better than the default `grpc.GzipCompressor`.

```bash
> go test -bench "^Benchmar.*(Compressor|Decompressor)$" -run "^$"
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/br/pkg/lightning/backend/local
cpu: AMD Ryzen 7 3700X 8-Core Processor             
BenchmarkGzipCompressor-16                  2149            698146 ns/op
BenchmarkGrpcGzipCompressor-16                60          20362977 ns/op
BenchmarkGzipDecompressor-16                1005           1107029 ns/op
BenchmarkGrpcGzipDecompressor-16             986           1122713 ns/op
PASS
ok      github.com/pingcap/tidb/br/pkg/lightning/backend/local  6.545s
```

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/backend/local/local.go [hunk 1]
```diff
@@ -183,9 +180,13 @@ func (f *importClientFactoryImpl) makeConn(ctx context.Context, storeID uint64)
 	)
 	switch f.compressionType {
 	case config.CompressionNone:
-	// do nothing
+		// do nothing
 	case config.CompressionGzip:
-		opts = append(opts, grpc.WithDefaultCallOptions(grpc.UseCompressor(gzip.Name)))
+		// Use custom compressor/decompressor to speed up compression/decompression.
```
Comment (lance6716) [inline br/pkg/lightning/backend/local/local.go:185]: maybe we can register our gzip Compressor with the name "gzip2" or something? Not important, we can postpone it when upstream remove this API

**Review by lance6716 — APPROVED** [review]

**Review by lichunzhu — APPROVED** [review]
LGTM

File: br/pkg/lightning/backend/local/local.go (hunk 1 above)
Comment (gozssky) [reply br/pkg/lightning/backend/local/local.go:185]: No, we can't. The name will be sent to the server. The server can't recognize `gzip2`.

**Pull request merged**